runtime/jaktlib/prelude/operators.jakt

```diff
@@ -39,6 +39,14 @@ trait ThrowingDivide<Rhs, Output> {
     fn divide(this, anon rhs: Rhs) throws -> Output
 }
 
+trait Index<Output> {
+    fn index(this, anon index: usize) -> Output
+}
+
+trait ThrowingIndex<Output> {
+    fn index(this, anon index: usize) throws -> Output
+}
```

Comment:
We have indexing with non-usize keys (for dictionaries), how about:
```suggestion
trait Index<Key, Output> {
    fn index(this, anon index: Key) -> Output
}

trait ThrowingIndex<Key, Output> {
    fn index(this, anon index: Key) throws -> Output
}
```

and following that,
```
type Array implements(Index<usize, T>) {
    [[name="operator[]"]]
    extern fn index(this, anon index: usize) -> T
}

type Dictionary implements(Index<K, V>) {
    [[name="operator[]"]]
    extern fn index(this, anon index: K) -> V
}
```

Reply:
Eh, codegen puts parameter in the wrong place
 ```cpp
 ((((test).values))[](static_cast<i64>(1LL))));
 ```
 let's see how to fix that...

Reply:
with `operator[]`? it should be calling `.operator[](...)` with that :thinking:

Reply:
ah, I did `operator([])` by mistake